### ISSUE TITLE
Stop relying implicit Bacon.js stream ordering and update library

### ISSUE THREAD

**Issue by maurizi** [OpenTreeMap/otm-core#3129]
I attempted to update Bacon.js as part of https://github.com/OpenTreeMap/otm-core/pull/3128, but updating caused regressions in our search handling code (as well as possibly other places).

The regressions are due to a reliance on the implicit ordering of calls to `onValue` that changed from one version to another.  One `onValue` was storing data in the DOM as a side-effect, and a `filter` in another stream was attempting to read that data.  The proper way to fix this is to ensure that all data dependencies of streams are captured by the stream itself.

**Comment by RobinIsTheBird** [comment]
That could explain part of why adding any new component to the search Bacon is hard. Glad to see this as an issue. Is there another issue for updating Bacon that is blocked on this one?
